### PR TITLE
docs(health-check): more notes

### LINKED ISSUE #4102: Passive healthcheck not working when a route has only one upstream
### Environment
* apisix version: 2.0
* OS: 3.10.0-693.el7.x86_64
* OpenResty version: openresty/1.17.8.2
* etcd version: 3.4.13
* apisix-dashboard version: 2.0

### Minimal test code / Steps to reproduce the issue
1. Set a route with only one upstream. Add active and passive healthcheck.
2. Send requests.

### What's the actual result? 
Even if this upstream always return 500, the passive healthcheck does not work to set this node to be unhealthy. Every request will still be passed to this upstream. 
It seems that the checker is not created at all, even after I set this upstream return 200.

The route config in etcd:
```
{
	"id": "route_2050210843",
	"create_time": 1618455175,
	"update_time": 1619011016,
	"uris": ["/test1.do"],
	"name": "route-2050210843",
	"desc": "health_check_test",
	"methods": ["GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS"],
	"vars": [],
	"upstream": {
		"nodes": [{
			"host": "172.25.69.182",
			"port": 8080,
			"weight": 1
		}],
		"timeout": {
			"connect": 6,
			"read": 6,
			"send": 6
		},
		"type": "roundrobin",
		"checks": {
			"active": {
				"healthy": {
					"interval": 100,
					"successes": 10
				},
				"host": "172.25.69.182",
				"http_path": "/check.do",
				"req_headers": [""],
				"timeout": 3,
				"unhealthy": {
					"http_failures": 5,
					"interval": 3
				}
			},
			"passive": {
				"healthy": {
					"http_statuses": [200],
					"successes": 2
				},
				"unhealthy": {
					"http_failures": 1,
					"http_statuses": [500, 502, 504],
					"tcp_failures": 1
				}
			}
		}
	},
	"labels": {
		"serve": "2050210843"
	}
}
```

### What's the expected result?
The checker for this upstream is created and can be set to unhealthy state when failures reach passive.unhealthy.http_failures.

### PR DESCRIPTION
**Pull request by spacewander**
Fix #4102 
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by dabue — APPROVED** [review]
LGTM

File: docs/en/latest/health-check.md [hunk 1]
```diff
@@ -26,8 +26,13 @@ title: Health Check
 Health Check of APISIX is based on [lua-resty-healthcheck](https://github.com/Kong/lua-resty-healthcheck),
 you can use it for upstream.
 
-Note that we only start the health check when the upstream is hit by a request.
+Note:
+
+* We only start the health check when the upstream is hit by a request.
 There won't be any health check if an upstream is configured but isn't in used.
+* If there is not healthy node can be chosen, we will continue to access the upstream.
```
Comment (tokers) [inline docs/en/latest/health-check.md:33]:
```suggestion
* If there is no healthy node can be chosen, we will continue to access the upstream.
```

File: docs/en/latest/health-check.md [hunk 2]
```diff
@@ -26,8 +26,13 @@ title: Health Check
...
+* We won't start the health check when the upstream only has one node, as we will access
+it whether this unique node is healthy or not.
```
Comment (tokers) [inline docs/en/latest/health-check.md:35]:
```suggestion
it no matter this unique node is healthy or not.
```

Comment (spacewander) [reply docs/en/latest/health-check.md:35]: `whether ... or not` is equal to `no matter`.

**Review by tokers — APPROVED** [review]

**Pull request merged**